src/components/QuestionPageComponents/UploadJsonButton.tsx

```diff
@@ -0,0 +1,128 @@
+}
+
+const api = axios.create({
+    baseURL: "/* TODO */", // TODO
+    timeout: 5000,
+})
+
+const UploadJsonButton = () => {
+    const [open, setOpen] = React.useState(false);
+    const [file, setFile] = React.useState<File | null>(null);
+
+    const handleOpen = () => setOpen(true);
+    const handleClose = () => setOpen(false);
+
+    function handleAxiosError(error: any) {
+        console.error('An error occurred:', error.message);
+        return null;
+    }
+
+    const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        if (e.target.files) {
+            setFile(e.target.files[0]);
+        }
+    };
+
+    const handleUpload = async () => {
+        const formData = new FormData();
+        if (file) {
+            formData.append('file', file);
+        }
+        console.log(formData);
+
+        try {
+            const response = await api.post("/", formData);
+            handleClose();
+            return response.data;
+        } catch (error) {
+            console.log(file?.name);
+            handleAxiosError(error);
+        }
+    }
+
+    return (
+        <Box>
+          <Button
+                variant="contained"
+                color="secondary"
+                startIcon={<UploadFileIcon />}
+                sx={{ color: 'white' }}
+                onClick={handleOpen}
+            >
+            Upload JSON
+            </Button>
+
+            <Modal
+                open={open}
+                onClose={handleClose}
+                aria-labelledby="modal-title"
+                aira-describedby="modal-description"
+            >
+                <Box
+                    sx={{
+                        position: 'absolute',
+                        top: '50%',
+                        left: '50%',
+                        transform: 'translate(-50%, -50%)',
+                        width: 1050,
+                        bgcolor: 'background.paper',
+                        border: '2px solid #000',
+                        boxShadow: 24,
+                        p: 4,
+                        borderRadius: 2,
+                    }}
+                >
+                    <Typography id="modal-title" variant="h6" component="h2" sx={{ color: 'white' }}>
+                        Upload JSON file
+                    </Typography>
+                    <Typography variant="subtitle2" sx={{ color: 'white', mt: 2 }}>
+                        Example JSON file
+                    </Typography>
+                    <CopyBlock
```

Comment:
On my screen, the UI looks too big. 

<img width="1437" alt="Screenshot 2024-09-23 at 9 46 30 PM" src="https://github.com/user-attachments/assets/b626275d-ae26-4a05-aea3-251ac570bd55">

Propose changes to code (Adds a horizontal scroll bar if needed, reduce fontsize to 12px):
`<div style={{ overflowX: 'auto', maxWidth: '100%', fontSize: "12px" }}>
                        <CopyBlock
                            language={codeBlockProps.language}
                            showLineNumbers={codeBlockProps.showLineNumbers}
                            wrapLongLines={false}
                            theme={codeBlockProps.theme}
                            text={codeBlockProps.code}
                        />
                    </div>`

Reply:
Changed accordingly by adding a horizontal scroll bar and adjusted the fontsize of codeblock and width of modal.